### PR TITLE
Implement feature to allow user to chat with its friends

### PR DESCRIPTION
**Pull request by MarcelTorne**
This PR is bringing friend-to-friend chatting. This feature is available when going to the friends list through the profile and clicking on a friend to chat with him/her. 

This PR is also fixing a bug that was in the MapActivity. When we launched the map activity without permissions and we accepted the permissions the user wasn't centred and the clusterer for displaying the POIs was not initialised. This has been fixed now.

### CODE REVIEW

**Review by Bwananah — APPROVED** [review]
Looks good ! Good job !

**Comment by tberkane** [comment]
I'd like to test this but I don't have a friend, would it be possible to add a default friend?

**Comment by MarcelTorne** [comment]
> I'd like to test this but I don't have a friend, would it be possible to add a default friend?

Thanks for the suggestion, I added the Meili user as a default friend.

File: app/src/main/res/layout/fragment_chat_log.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    android:layout_height="match_parent"
+    tools:context=".messages.ChatLogFragment">
+
+    <!-- TODO: Update blank fragment layout -->
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:text="@string/hello_blank_fragment" />
+
+</FrameLayout>
```
Comment (aman3014) [inline app/src/main/res/layout/fragment_chat_log.xml:14]: Just wondering: what is this a placeholder for?

File: app/src/main/java/com/github/epfl/meili/profile/friends/FriendsListRecyclerAdapter.kt [hunk 2]
```diff
@@ -25,6 +28,13 @@ class FriendsListRecyclerAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>
     }
 
     class FriendViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        init {
+            itemView.setOnClickListener{
+                val intent = Intent(MainApplication.applicationContext(), ChatLogActivity::class.java)
+                MainApplication.applicationContext().startActivity(intent)
+            }
+        }
```
Comment (aman3014) [inline app/src/main/java/com/github/epfl/meili/profile/friends/FriendsListRecyclerAdapter.kt:36]: What is this for?

**Review by aman3014 — COMMENTED** [review]
Looks good, I just had a couple of questions before I approve.

File: app/src/main/java/com/github/epfl/meili/profile/friends/FriendsListRecyclerAdapter.kt (hunk 2 above)
Comment (MarcelTorne) [reply app/src/main/java/com/github/epfl/meili/profile/friends/FriendsListRecyclerAdapter.kt:36]: Sorry this was one way I tried to add transition to the chat activity but forgot to remove it. Thanks for the catch!

File: app/src/main/res/layout/fragment_chat_log.xml (hunk 1 above)
Comment (MarcelTorne) [reply app/src/main/res/layout/fragment_chat_log.xml]: Removed, same problem as before. Thanks for catching it!

**Review by aman3014 — APPROVED** [review]
Looks good to go, thanks for this feature 😄

**Pull request merged**